packages/core/src/events/interfaces.ts

```diff
@@ -38,71 +38,137 @@ export interface CoreOptions {
      */
     path?: string
+    /**
+     * @example https://www.foo.com/
+     */
     referrer?: string
+    /**
+     * @example projectId=123
+     */
     search?: string
+    /**
+     * @example Analytics Academy
+     */
     title?: string
+    /**
+     * @example https://segment.com/academy/
+     */
     url?: string
   }
 
   /**
-   *  {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L285}
+   * User agent of the device making the request.
+   * (node only, collected automatically in browser).
    */
   userAgent?: string
 
   /**
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L286-L289}
-   */
-  locale?: string
-
-  /**
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L290-L291}
+   * Information about the SDK. Should not be filled out by the user.
+   * This type should probably be "never"
    */
   library?: {
+    /**
+     * @example analytics-node-next/latest
+     */
     name: string
+    /**
+     * @example  "1.43.1"
+     */
     version: string
   }
 
   /**
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L292-L301}
+   * This is useful in cases where you need to track an event,
+   * but also associate information from a previous identify call.
+   * You should fill this object the same way you would fill traits in an identify call.
    */
   traits?: {
     crossDomainId?: string
+    [customTrait: string]: any
   }
 
   /**
-   * utm params
-   * {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L303-L305}
-   * {@link https://github.com/segmentio/utm-params/blob/master/lib/index.js#L49}
+   * Dictionary of information about the campaign that resulted in the API call, containing name, source, medium, term, content, and any other custom UTM parameter.
    */
   campaign?: {
-    /**
-     * This can also come from the "utm_campaign" param
-     *
-     * {@link https://github.com/segmentio/utm-params/blob/master/lib/index.js#L40}
-     */
     name: string
     term: string
     source: string
     medium: string
     content: string
+    [customUTMParam: string]: any
   }
 
   /**
-   *  {@link https://github.com/segmentio/analytics.js-integrations/blob/2d5c637c022d2661c23449aed237d0d546bf062d/integrations/segmentio/lib/index.js#L415}
+   * Dictionary of information about the way the user was referred to the website or app.
    */
   referrer?: {
-    btid?: string
-    urid?: string
+    type?: string
+    name?: string
+    url?: string
+    link?: string
+
+    btid?: string // TODO? where did this come from?
```

Comment:
have no idea where this came from -- it's not in https://segment.com/docs/connections/spec/common/#context